MobileSdk/src/main/java/com/spruceid/mobile/sdk/GattClient.kt

```diff
@@ -333,101 +348,300 @@ class GattClient(private var callback: GattClientCallback,
         override fun onCharacteristicChanged(gatt: BluetoothGatt,
                                              characteristic: BluetoothGattCharacteristic) {
 
-            callback.onLog("onCharacteristicChanged, uuid=${characteristic.uuid}")
+            reportLog("onCharacteristicChanged, uuid=${characteristic.uuid}")
             @Suppress("deprecation")
             val value = characteristic.value
-            if (characteristic.uuid.equals(characteristicServer2ClientUuid)) {
-                if (value.isEmpty()) {
-                    callback.onError(Error("Invalid data length ${value.size} for Server2Client " +
-                            "characteristic."))
-                    return
-                }
+            
+            when(characteristic.uuid) {
 
-                Log.d("GattClient.onCharacteristicChanged", byteArrayToHex(value))
+                characteristicServer2ClientUuid -> {
+                    if (value.isEmpty()) {
+                        reportError("Invalid data length ${value.size} for Server2Client characteristic.")
+                        return
+                    }
 
-                incomingMessage.write(value, 1, value.size - 1)
+                    Log.d("GattClient.onCharacteristicChanged", byteArrayToHex(value))
```

Comment:
probably replace it with the newly introduced `reportLog`

Reply:
I've removed it; it was debug scaffolding.